core/src/main/scala/scalafix/util/OrganizeImports.scala

```diff
@@ -0,0 +1,200 @@
+package scalafix.util
+
+import scala.collection.immutable.Seq
+import scala.collection.mutable
+import scala.meta.Importee.Wildcard
+import scala.meta._
+import scala.meta.tokens.Token.Comment
+import scalafix.FilterMatcher
+import scalafix.rewrite.RewriteCtx
+import scalafix.syntax._
+import scalafix.util.TreePatch.AddGlobalImport
+import scalafix.util.TreePatch.RemoveGlobalImport
+
+object OrganizeImports {
+  def extractImports(stats: Seq[Stat])(
+      implicit ctx: RewriteCtx): (Seq[Import], Seq[CanonicalImport]) = {
+    val imports = stats.takeWhile(_.is[Import]).collect { case i: Import => i }
+    val importees = imports.collect {
+      case imp @ Import(importers) =>
+        implicit val currentImport = imp
+        importers.flatMap { importer =>
+          val ref = importer.ref //  fqnRef.getOrElse(importer.ref)
+          val wildcard = importer.importees.collectFirst {
+            case wildcard: Importee.Wildcard => wildcard
+          }
+          wildcard.fold(importer.importees.map(i => CanonicalImport(ref, i))) {
+            wildcard =>
+              val unimports = importer.importees.collect {
+                case i: Importee.Unimport => i
+              }
+              val renames = importer.importees.collect {
+                case i: Importee.Rename => i
+              }
+              List(CanonicalImport(ref, wildcard, unimports, renames))
+          }
+        }
+    }.flatten
+    imports -> importees
+  }
+
+  def getLastTopLevelPkg(potPkg: Stat): Stat = potPkg match {
+    case Pkg(_, head +: Nil) => getLastTopLevelPkg(head)
+    case Pkg(_, head +: _) => head
+    case _ => potPkg
+  }
+
+  def getGlobalImports(ast: Tree)(
+      implicit ctx: RewriteCtx): (Seq[Import], Seq[CanonicalImport]) =
+    ast match {
+      case Pkg(_, Seq(pkg: Pkg)) => getGlobalImports(pkg)
```

Comment:
This whole file is great, I love how readable you have made it. I did one read through and immediately understood the code and how to use it 👍